src/main/java/com/feedzai/commons/sql/abstraction/dml/result/MySqlResultColumn.java

```diff
@@ -32,4 +32,14 @@
     public MySqlResultColumn(final String name, final Object val) {
         super(name, val);
     }
+
+    @Override
+    public Boolean toBoolean() {
+        try {
+            return super.toBoolean();
+        } catch (ClassCastException e) {
+            // Since booleans are represented using unsigned integers, we need to convert them.
```

Comment:
AFAIK the `toBoolean()` is already being called when I read a record from a table with a boolean column.  
Why was this method not required before?